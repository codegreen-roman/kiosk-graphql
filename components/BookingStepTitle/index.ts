import { FC, ReactNode } from 'react'
import { chooseThemeComponent } from '@utils/theming'
import themeNames from '@const/themeNames'
import KihnuKiosk from './themes/kihnuKiosk/BookingStepTitle'

export interface BookingStepTitleProps {
  children: ReactNode
  totalAmount?: number
  finalStep?: string
}

export const BookingStepTitle: FC<BookingStepTitleProps> = chooseThemeComponent({
  [themeNames.kihnuKiosk]: KihnuKiosk,
})
