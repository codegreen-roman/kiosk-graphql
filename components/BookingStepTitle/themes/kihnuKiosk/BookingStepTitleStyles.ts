import styled from 'styled-components'
import { getFonts, getPageHeadingColors, themeSettings } from '@themeSettings'
import { Container as ContainerComponent } from '@material-ui/core'

const pageHeadingColors = getPageHeadingColors()
const { buttons } = themeSettings.colors
const { primary, secondary } = getFonts()

export const Root = styled.div<{ totalAmount: number; finalStep?: string }>`
  width: 100%;
  background: ${(props): string => {
    return props.totalAmount >= 0
      ? `linear-gradient(90deg, ${pageHeadingColors.mainBackground} 50%, ${pageHeadingColors.extraBackground} 50%)`
      : pageHeadingColors.mainBackground
  }};
`

export const Container = styled(ContainerComponent)`
  position: relative;
  display: flex !important;
  align-items: center;
  justify-content: start;
  width: 100%;
`

export const LeftSide = styled.div<{ finalStep?: string }>`
  background-color: ${pageHeadingColors.mainBackground};
  font-family: ${secondary};
  font-weight: 600;
  font-size: 50px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  color: ${pageHeadingColors.mainText};
  display: flex;
  align-items: center;
  flex-grow: 0;
  flex-basis: 75%;
  max-width: 75%;
  min-height: 80px;
  margin-left: ${(props): string => {
    return props.finalStep ? '24px' : '0px'
  }};
`

export const RightSide = styled.div<{ finalStep?: string }>`
  display: flex;
  justify-content: ${(props): string => {
    return props.finalStep ? 'center' : 'space-between'
  }};
  align-items: center;
  flex-grow: 0;
  max-width: 25%;
  flex-basis: 25%;
  min-height: 80px;
  margin-right: ${(props): string => {
    return props.finalStep ? '10px' : '0px'
  }};
`

export const TotalAmountLabel = styled.div`
  font-family: ${primary};
  font-size: 20px;
  font-weight: 600;
  color: ${pageHeadingColors.label};
  text-align: right;
  margin: 0 22px;
  max-width: 167px;
`

export const FinalStepLabel = styled.div`
  font-family: ${secondary};
  text-transform: uppercase;
  font-size: 25px;
  font-weight: 500;
  letter-spacing: 0.75px;
  color: ${buttons.button.defaultText};
  text-align: right;
  max-width: 167px;
`

export const TotalAmount = styled.div`
  font-family: ${secondary};
  font-weight: 600;
  font-size: 50px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  color: ${pageHeadingColors.extraText};
  flex-shrink: 0;
`

export const FinishBox = styled.div`
  background-color: ${buttons.button.cancelBackground};
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 10px 0 11px 0;
  min-height: 80px;
`
