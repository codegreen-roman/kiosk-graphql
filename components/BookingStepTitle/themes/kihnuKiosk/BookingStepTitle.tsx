import React, { FC } from 'react'
import { asMoneyString } from '@utils/formatters'
import { BookingStepTitleProps } from '@components/BookingStepTitle'
import {
  Root,
  Container,
  LeftSide,
  RightSide,
  TotalAmount,
  TotalAmountLabel,
  FinalStepLabel,
  FinishBox,
} from './BookingStepTitleStyles'
import { useTranslation } from 'react-i18next'
import { AppStateTypes, useAppState } from '../../../../hooks/useAppState'
import CircularProgress from '@material-ui/core/CircularProgress'
import { Box } from '@material-ui/core'

const BookingStepTitle: FC<BookingStepTitleProps> = ({ totalAmount, children, finalStep }) => {
  const { t } = useTranslation()
  const { dispatch } = useAppState()
  const handleRedirect = (): void => {
    dispatch({ type: AppStateTypes.startAgain, payload: true })
  }
  return (
    <Root totalAmount={totalAmount} finalStep={finalStep}>
      <Container maxWidth={finalStep ? false : 'xl'} disableGutters={!!finalStep}>
        <LeftSide finalStep={finalStep}>{children}</LeftSide>
        <RightSide display-if={totalAmount !== undefined}>
          <TotalAmountLabel>{t('Total price label')}</TotalAmountLabel>
          {totalAmount >= 0 ? (
            <TotalAmount>{asMoneyString(totalAmount / 100)}</TotalAmount>
          ) : (
            <Box display="flex" alignItems="center" justifyContent="center" marginRight="15%">
              <CircularProgress color="primary" />
            </Box>
          )}
        </RightSide>
        <RightSide display-if={Boolean(finalStep)} finalStep={finalStep}>
          <FinishBox onClick={handleRedirect}>
            <FinalStepLabel>{t('Finish button')}</FinalStepLabel>
          </FinishBox>
        </RightSide>
      </Container>
    </Root>
  )
}
export default BookingStepTitle
