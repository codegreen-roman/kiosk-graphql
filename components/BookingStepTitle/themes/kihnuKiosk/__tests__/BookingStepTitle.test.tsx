import React from 'react'
import { LeftSide, TotalAmount } from '../BookingStepTitleStyles'
import { shallow } from 'enzyme'

jest.mock('react-i18next', () => ({
  useTranslation: (): object => ({
    t: (): string => 'Total price label',
  }),
}))

jest.mock('../../../../../hooks/useAppState', () => ({
  useAppState: (): object => ({
    dispatch: (): void => {},
  }),
}))

import { BookingStepTitle } from '@components/BookingStepTitle'

describe('BookingStepTitle', () => {
  const component = shallow(<BookingStepTitle totalAmount={0}>Title</BookingStepTitle>)

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders children as left side', () => {
    expect(component.find(LeftSide)).toHaveText('Title')
  })

  it('renders right total amount with currency sign', () => {
    expect(component.find(TotalAmount)).toHaveText('0.00 €')
  })
})
