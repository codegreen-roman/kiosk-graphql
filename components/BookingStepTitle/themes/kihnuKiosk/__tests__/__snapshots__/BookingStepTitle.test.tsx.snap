// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingStepTitle should match snapshot 1`] = `
<BookingStepTitleStyles__Root
  totalAmount={0}
>
  <BookingStepTitleStyles__Container
    disableGutters={false}
    maxWidth="xl"
  >
    <BookingStepTitleStyles__LeftSide>
      Title
    </BookingStepTitleStyles__LeftSide>
    <BookingStepTitleStyles__RightSide>
      <BookingStepTitleStyles__TotalAmountLabel>
        Total price label
      </BookingStepTitleStyles__TotalAmountLabel>
      <BookingStepTitleStyles__TotalAmount>
        0.00 €
      </BookingStepTitleStyles__TotalAmount>
    </BookingStepTitleStyles__RightSide>
  </BookingStepTitleStyles__Container>
</BookingStepTitleStyles__Root>
`;
